### PR TITLE
Fix interactive client with older server

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Not for changelog

Fix interactive client with older server after https://github.com/ClickHouse/ClickHouse/pull/38970
The problem was that older servers don't know the function `viewIfPermitted()` and thus can't make a suggest for a newer client.

### CODE REVIEW

**Review by Avogar — APPROVED** [review]

**Comment by Avogar** [comment]
This PR is needed for release, the only failure from the first commit (bad style of variable naming) was fixied in the second commit, so let's merge

File: src/Client/Suggest.h [hunk 1]
```diff
@@ -28,8 +28,8 @@ class Suggest : public LineReader::Suggest, boost::noncopyable
     template <typename ConnectionType>
     void load(ContextPtr context, const ConnectionParameters & connection_parameters, Int32 suggestion_limit);
 
-    /// Older server versions cannot execute the query above.
-    static constexpr int MIN_SERVER_REVISION = 54406;
+    /// Older server versions cannot execute the query loading suggestions.
+    static constexpr int MIN_SERVER_REVISION = DBMS_MIN_PROTOCOL_VERSION_WITH_VIEW_IF_PERMITTED;
```
Comment (azat) [inline src/Client/Suggest.h:32]: @vitlibar maybe it worth fixing in a more appropriate way, by allowing autocompletion compatibility too, i.e. by wrapping with `viewIfPermitted` only if `server_revision >= DBMS_MIN_PROTOCOL_VERSION_WITH_VIEW_IF_PERMITTED`, what do you think?

Comment (lzydmxy) [reply src/Client/Suggest.h:32]: > @vitlibar maybe it worth fixing in a more appropriate way, by allowing autocompletion compatibility too, i.e. by wrapping with `viewIfPermitted` only if `server_revision >= DBMS_MIN_PROTOCOL_VERSION_WITH_VIEW_IF_PERMITTED`, what do you think?

I think so. We should be compatible with older versions. I got a lot clickhouse servers with different versions, however, the latest version client can't give any suggestions when connect to older version server. This brings me great inconvenience to use I am going to fix it, when server_revision < DBMS_MIN_PROTOCOL_VERSION_WITH_VIEW_IF_PERMITTED, we should chose other way to load suggestions.

Comment (azat) [reply src/Client/Suggest.h:32]: Any progress on this one? It is OK not to fix it now I guess, but there is one more thing that breaks completion, so I remembered about this again - https://github.com/ClickHouse/ClickHouse/pull/62000/files#r1583289769.

**Pull request merged**